### PR TITLE
Allow custom folder for SDMC and NAND Directories

### LINKED ISSUE #3818: Suggestion: Allow customizing save file directory
<-- not an emulator issue, merely a suggestion and nowhere to put it -->

Other emulators (such as Dolphin) allow you to choose a custom directory for game saves. This works well for cases like Dropbox that syncs a folder in %AppData% between any devices logged in with that Dropbox account. There are many other uses for custom save directories but that is my usage. This can currently be fixed by using symbolic links but the feature would be a nice addition regardless.

### PR DESCRIPTION
**Pull request by nieldm**
This attempts to Fix #3818

Moved the configuration to the config dialog now You can See, Change and Open the NAND and the SDMC directory

CustomFolders
Default
GameRunning
VirtualSDDisable

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5759)

### CODE REVIEW

**Comment by B3n30** [comment]
Thx for the commit.

here are some remarks: 
1. UI changes require some screenshots. 
2. I dont think this should be configurable from the menu, But should be Part of the config dialog
3. Changing those folders should only be possible, when the emulation isnt running or Citra might crash
4. There needs to be some concept about save states, or we might introduce some issues here

**Comment by nieldm** [comment]
Thx for feedback @B3n30 I will do the changes to cover this points, not sure about the last one.

Just picked up this ticket to start diving into citra's code.

File: src/citra_qt/configuration/config.cpp [hunk 1]
```diff
@@ -302,6 +302,16 @@ void Config::ReadDataStorageValues() {
     qt_config->beginGroup(QStringLiteral("Data Storage"));
 
     Settings::values.use_virtual_sd = ReadSetting(QStringLiteral("use_virtual_sd"), true).toBool();
+    std::string nan_dir = FileUtil::GetUserPath(FileUtil::UserPath::NANDDir);
+    Settings::values.nand_dir =
+        ReadSetting(QStringLiteral("nand_directory"), QString::fromStdString(nan_dir))
```
Comment (JohelEGP) [inline src/citra_qt/configuration/config.cpp:307]:
```suggestion
    std::string nand_dir = FileUtil::GetUserPath(FileUtil::UserPath::NANDDir);
    Settings::values.nand_dir =
        ReadSetting(QStringLiteral("nand_directory"), QString::fromStdString(nand_dir))
```

**Review by JohelEGP — CHANGES_REQUESTED** [review]
I haven't tried this build, only reviewed the code. I notice that sometimes NAND comes first, others SDMC, for otherwise almost equal code.

**Comment by nieldm** [comment]
> Thx for the commit.
> 
> here are some remarks:
> 
> 1. UI changes require some screenshots.
> 2. I dont think this should be configurable from the menu, But should be Part of the config dialog
> 3. Changing those folders should only be possible, when the emulation isnt running or Citra might crash
> 4. There needs to be some concept about save states, or we might introduce some issues here

I just updated the branch solving points 1, 2 and 3, but 4 I feel it's out of my current knowledge of Citra, but I will be glad to learn more about it, could you give me more context and/or point me in the right direction.

**Review by B3n30 — APPROVED** [review]
Looks good to me now. Thanks for the good work.

For 4.: I think #5760 is a big step in the right direction, to solve this issue. Maybe you have a look at it, to see if it is compatible with your changes.

**Comment by nieldm** [comment]
> #5760

Perfect thx I would take look into it

**Review by bunnei — APPROVED** [review]

**Comment by zhaowenlan1779** [comment]
Obviously I am too late, but I have to say ... I hate those two lines. The captialization on the "T" and the period are missing.

**Pull request merged**